File: problems/ABC/ABC107/C.py
N, K = map(int, input().split())
x = list(map(int, input().split()))
ans = 10**18
if K == 1:
    for i in range(N):
        ans = min(ans, abs(x[i]))
else:
    start = 0
    shift = min(x)
    if shift < 0:
        start = abs(shift)
        for i in range(N):
            x[i] += abs(shift)

    l = 0
    for r in range(K - 1, N):
        dist = x[r] - x[l]
        if x[l] <= start <= x[r]:
            ans = min(ans, dist + min(start - x[l], x[r] - start))
        elif start < x[l]:
            ans = min(ans, dist + (x[l] - start))
        elif x[r] < start:
            ans = min(ans, dist + (start - x[r]))
        l += 1
print(ans)

File: problems/ABC/ABC107/C.cpp
#include <bits/stdc++.h>
using namespace std;

using ll = long long;

#define rep(i, a, b) for(int i = a; i < b; i++)
#define rrep(i, a, b) for(int i = a; i >= b; i--)
#define fore(i, a) for(auto &i : a)

#define all(x) (x).begin(),(x).end()

int main(){
    ll N, K;
    cin >> N >> K;
    vector<ll> x(N);
    rep(i, 0, N){
        cin >> x[i];
    }
    ll ans = 1 << 30;
    if(K == 1){
        rep(i, 0, N){
            ans = min(ans, abs(x[i]))
        }
    } else{
        vector<ll> s(n)
        int l = 0;
        ll a = 0;
        for(int r = 1; r < N; r++){
            a += 
        }
    }

    
}